hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el

```diff
@@ -146,6 +146,135 @@ Condition 37 "2164803938" "(tl_i_int.a_valid & reqfifo_wready & ((~error_interna
+// goes to u_sram_byte as one of the port of tl_o_int and d_ready port of tl_i_int is true which
+// comes out from u_sram_byte. But d_valid to be true requires reqfifo_rvalid true.
+INSTANCE: tb.dut.u_tl_adapter_rom.u_reqfifo
+Condition 3 "1324655787" "(rvalid_o & rready_i & ((~gen_normal_fifo.under_rst))) 1 -1" (1 "011")
+
+// For u_rspfifo, Pass=1 and rvalid_o is ~(fifo_empty & ~wvalid_i) & ~under_rst. wvalid_i comes
+// from u_tl_adapter_rom as rspfifo_wvalid. rspfifo_wvalid needs both rvalid_i (response from rom)
+// and reqfifo_rvalid. If we want rready_i=1, we want reqfifo_rready=1 in u_tl_adapter_rom.
+// reqfifo_rready further gets assigned by tl_o_int.dvalid and tl_i_int.d_ready. For
+// tl_o_int.dvalid to set itself true requires reqfifo_rvalid. Which means that read from reqfifo
+// is happening and u_rspfifo needs to respond to that read on the same cycle. So, we will get
+// rvalid_o with rready_i.
+INSTANCE: tb.dut.u_tl_adapter_rom.u_rspfifo
+Condition 3 "1324655787" "(rvalid_o & rready_i & ((~gen_normal_fifo.under_rst))) 1 -1" (1 "011")
+
+// For u_sramreqfifo, the rready_i needs reqfifo_rvalid true in u_tl_adapter_rom. But this means
+// that we are reading the request we wrote earlier. If this is the case then we can't get rvalid_o
+// low as this means that the fifo is empty.
+INSTANCE: tb.dut.u_tl_adapter_rom.u_sramreqfifo
+Condition 3 "1324655787" "(rvalid_o & rready_i & ((~gen_normal_fifo.under_rst))) 1 -1" (1 "011")
+
+// The case 110 for the conditional statements below is non-occuring.
+//
+// To cover 110, all the bits of the conditional statements should be true. If under_rst is true,
+// then we get empty_o in u_fifo_cnt is true. empty_o=1 when wptr_wrap_cnt_o = rptr_wrapt_cnt_o in
+// u_fifo_cnt. As Secure is true in case of rom_ctrl, wptr_wrap_cnt_o and rptr_wrap_cnt_o comes out
+// as cnt_o from u_wptr and u_rptr respectively. Both u_wptr and u_rptr are instances of same
+// module (prim_count). Then cnt_o gets assigned by cnt_q[0] which is updated as a result of
+// fpv_force[0] + cnt_unforced_q. fpv_force is wired to 0 as rom_ctrl doesn't define the macro
+// PrimCountFPV. cnt_unforced_q comes out as q_o from u_cnt_flop. In u_cnt_flop, q_o comes out from
+// u_impl_generic instance which contains non-blocking assignments to q_o. If !rst_ni, then q_o
+// will get assigned by a ResetValue which is 0 in this case. Otherwise, q_o will be delayed by
+// d_i. This will get cnt_q=0 in both u_ptr and w_ptr which leads to cnt_o=0. When cnt_o in both
+// w_ptr and r_ptr is 0 leads wptr_wrap_cnt_q and rptr_wrap_cnt_q to be equal. Which means the
+// empty_o=1 in u_fifo_cnt and it will be seen as fifo_empty in u_**fifo.
+//
+// For u_reqfifo, Pass=0 so fifo_empty is rvalid_o. Hence, we can't see rvalid_o when under_rst is
+// true. With no rvalid_o, we can't get rready_i and the reasoning above for case 011 for the same
+// conditional statement explains why.
```

Comment:
I think there's a rather shorter chain of logic that gets here:
```systemverilog
// We can never see rvalid_o && under_rst in u_reqfifo in any tlul_adapter_sram instance. Since the
// Pass parameter is false for u_reqfifo, rvalid_o is only true if the FIFO is nonempty. Since the
// FIFO clears on reset, this cannot be true in the first cycle after reset. But that is the only
// cycle when under_rst is true.
```